tool/tctl/common/plugins_command.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+
+	pluginsv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/plugins/v1"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+)
+
+// PluginsCommand allows for management of plugins.
+type PluginsCommand struct {
+	config     *servicecfg.Config
+	cleanupCmd *kingpin.CmdClause
+	pluginType string
+	dryRun     bool
+}
+
+// Initialize creates the plugins command and subcommands
+func (p *PluginsCommand) Initialize(app *kingpin.Application, config *servicecfg.Config) {
+	p.config = config
+
+	pluginsCommand := app.Command("plugins", "Manage Teleport plugins.").Hidden()
+	p.cleanupCmd = pluginsCommand.Command("cleanup", "Cleans up the given plugin type.")
+	p.cleanupCmd.Arg("type", "The type of plugin to cleanup.").StringVar(&p.pluginType)
```

Comment:
There is (unfortunately) no corresponding `enum` for this at the moment, and that ship has sort of sailed.

Reply:
Don't we have `typs.PluginTypeOkta`?

Reply:
We do, but it ends up sort of just being treated like a string everywhere. `PluginType` in `plugin_service.proto` is defined as:

```proto
// PluginType represents a single type of hosted plugin
// that can be onboarded.
message PluginType {
  // Type is a string corresponding to api.PluginTypeXXX constants
  string type = 1;

  // OAuthClientID contains the client ID of the OAuth application
  // that is used with this plugin's API provider.
  // For plugins that are not authenticated via OAuth,
  // this will be empty.
  string oauth_client_id = 2;
}
```

Reply:
I am confused now. What's the input to `--type`? Isn't `okta`? or does it also require the `oauth_client_id`?

If just `okta`, i think you can just cast `string(types.PluginTypeOkta)` for `EnumVar`.

Reply:
Ahhh, I see it now. Okay, will use `EnumVar`.

The PluginType in proto is confusing things here, sorry. I didn't realize the way `kingpin`'s `EnumVar` call worked. :-)